.github/workflows/pr-codeql-analysis-go.yml

```diff
@@ -0,0 +1,34 @@
+name: "CodeQL for PR / go"
+
+on:
+  pull_request:
+    branches: [main]
+    paths:
+      - '**/*.go'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: ['go']
```

Comment:
Nit. When running each language on it's own might not need to use the matrix feature? Do we want to fail fast in PR's? Maybe we don't